docs/adr/0016-use-graphql-and-apollo-for-settings-redesign.md

```diff
@@ -0,0 +1,135 @@
+# Use GraphQL and Apollo for Settings Redesign
+
+- Deciders: Lauren Zugai, Ben Bangert, Wil Clouser, Jody Heavener, Les Orchard, Danny Coates, Barry Chen, Vijay Budhram, Dave Justice, Alex Davis
+- Date: 2020-05-05
+
+## Context and Problem Statement
+
+The [Settings Redesign project](https://jira.mozilla.com/browse/FXA-840) will be created [as a new React application](https://github.com/mozilla/fxa/blob/master/docs/adr/0011-create-new-react-app-for-settings-redesign.md) and in turn, has opened the door to assess certain pieces of our technology stack.
+
+[GraphQL](https://graphql.org/) is not a database query language, but is instead a query language for APIs. It _describes_ data requirements, and is a powerful alternative to REST. Some benefits can be gained by using it on top of existing REST architecture.
+
+"Apollo" in this document refers to the "Apollo client" and "Apollo server" pieces of the Apollo platform¹, which can be described as a unified data layer that enables applications to interact with data from data stores and APIs. In other words, it allows us to use write and handle GraphQL on the client and server. Apollo also gives us many tools out of the box like caching.
+
+This ADR serves to lay out pros and cons of using GraphQL and Apollo in the Settings Redesign project as an alternative to hitting our conventional REST endpoints.
+
+¹Apollo also offers Apollo Graph Manger and Apollo Federation which are paid services, [read more from their docs](https://www.apollographql.com/docs/intro/platform/). We do not need to use these to use GQL and Apollo server and client.
```

Comment:
Could be "to use _the_ GQL and Apollo server and client", or possibly "to use the GQL server and Apollo client"?

Reply:
Hmm well, GQL isn't Apollo, but I'll rephrase to, "to use GQL with Apollo server or Apollo client" for clarity.